packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -260,6 +261,53 @@ interface UploadDataChunkParams<T extends UploadDataLookup, K extends keyof T> {
   batch: T[K][];
 }
 
+type HandleUploadDataChunkErrorParams = {
+  err: any;
+  attemptContext: AttemptContext;
+  logger: IntegrationLogger;
+};
+
+function handleUploadDataChunkError({
+  err,
+  attemptContext,
+  logger,
+}: HandleUploadDataChunkErrorParams): void {
+  if (err.code === 'RequestEntityTooLargeException') {
+    // No reason to retry these errors as the request size ain't gonna change.
+    throw new IntegrationError({
+      code: 'INTEGRATION_UPLOAD_FAILED',
+      cause: err,
+      fatal: false,
+      message: 'Failed to upload integration data because payload is too large',
+    });
+  } else if (err.code === 'JOB_NOT_AWAITING_UPLOADS') {
+    throw new IntegrationError({
+      code: 'INTEGRATION_UPLOAD_AFTER_JOB_ENDED',
+      cause: err,
+      fatal: true,
+      message:
+        'Failed to upload integration data because job has already ended',
+    });
+  }
+
+  if (
+    attemptContext.attemptsRemaining &&
+    // There are sometimes intermittent credentials errors when running
+    // a managed integration on AWS Fargate. They consistently succeed
+    // with retry logic, so we don't want to log a warn.
+    err.code !== 'CredentialsError'
+  ) {
+    logger.warn(
+      {
+        err,
+        code: err.code,
```

Comment:
Do you actually ever see a value for `err.code`?  I'm working in the same area for the work on shrinking rawData for `RequestEntityTooLargeException` error codes, but I never saw it as anything other than undefined.  I was looking at adding in checking for `err.response?.status` for the code, just because that's what I was used to using for Axios responses.  Would welcome any knowledge share on how this `err.code` is currently functioning.

Reply:
Good comms. This should be fixed in the latest commit.